src/main/java/bio/overture/ego/service/OAuthService.java

```diff
@@ -0,0 +1,94 @@
+package bio.overture.ego.service;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import bio.overture.ego.token.IDToken;
+
+@Service
+public class OAuthService {
+
+  @Value("${oauth.linkedIn.clientSecret}")
```

Comment:
You need to update the `application.yml` to at least have the proper config keys. Also any reason why you went with `oauth.linkedin` rather than just `linkedin` like google and facebook do?